scss/_patterns_code-snippet.scss

```diff
@@ -4,10 +4,46 @@ $color-snippet-heading-bg: rgba($color-x-dark, 0.08);
 
 @mixin vf-p-code-snippet {
   .p-code-snippet {
-    .p-code-snippet__block {
+    margin-bottom: $spv-outer--scaleable;
+
+    .p-code-snippet__block,
+    .p-code-snippet__block--icon {
       margin: 0;
     }
 
+    .p-code-snippet__block--icon {
+      padding-left: #{$sph-inner + map-get($icon-sizes, default) + $sph-inner--small};
+      position: relative;
+
+      &::before {
+        @extend %icon;
+        @include vf-icon-linux-prompt($color-mid-dark);
+
+        content: '';
+        left: $sph-inner;
+        position: absolute;
+        top: #{$spv-inner--x-small--scaleable + 0.5 * (map-get($line-heights, default-text) - map-get($icon-sizes, default))};
+        width: map-get($icon-sizes, default);
+
+        @if ($increase-font-size-on-larger-screens) {
+          // stylelint-disable-next-line max-nesting-depth
+          @media (min-width: $breakpoint-x-large) {
+            top: #{$spv-inner--x-small--scaleable + 0.5 * (map-get($line-heights, default-text) * $font-size-ratio--largescreen - map-get($icon-sizes, default))};
```

Comment:
@lyubomir-popov could you add a comment explaining a bit what this calculation does? It's a lot of variables

Reply:
$spv-inner--x-small--scaleable is the amount by which the parent is padded from the top, 
map-get($line-heights, default-text) * $font-size-ratio--largescreen is just the default line-hight at that media query 
o what's happening is we subtract the icon height from the lineheight and divide by two, this then gets added to the top offset. Not sure I can fit that in a comment